addons/mail/models/mail_channel.py

```diff
@@ -180,46 +211,54 @@ def _onchange_moderation(self):
 
     @api.model_create_multi
     def create(self, vals_list):
-        defaults = self.default_get(['image_128'])
-        current_partner = self.env.user.partner_id.id
+        defaults = self.default_get(['image_128', 'public'])
 
-        visibilities = []
+        access_types = []
         for vals in vals_list:
+            # find partners to add from partner_ids
+            partner_ids_cmd = vals.get('channel_partner_ids') or []
+            if any(cmd[0] == 0 for cmd in partner_ids_cmd):
+                raise ValidationError(_('Invalid value when creating a channel with members, only 4 or 6 are allowed.'))
+            partner_ids = [cmd[1] for cmd in partner_ids_cmd if cmd[0] == 4]
+            partner_ids += [cmd[2] for cmd in partner_ids_cmd if cmd[0] == 6]
+
+            # find partners to add from channel_last_seen_partner_ids
+            membership_ids_cmd = vals.get('channel_last_seen_partner_ids') or []
+            if any(cmd[0] == 0 for cmd in partner_ids_cmd):
```

Comment:
You're checking the wrong variable good sir (and maybe should be != instead of == according to error message? I'm so confused).

Reply:
Ouch. Such pâté :D .